torch/testing/_internal/distributed/nn/api/remote_module_test.py

```diff
@@ -385,6 +404,116 @@ def hook(module, grad_input, grad_output):
             ):
                 remote_module.extra_repr()
 
+    @dist_utils.dist_init
+    def test_send_remote_module_with_a_new_attribute_over_the_wire(self):
+        if self.rank != 0:
+            return
+        dst_worker_name = dist_utils.worker_name((self.rank + 1) % self.world_size)
+
+        # If add a new attribute is added to this RemoteModule, which will be sent over the wire by RPC,
+        # this new field must be added to either _REMOTE_MODULE_PICKLED_ATTRIBUTES or _REMOTE_MODULE_ATTRIBUTES_IGNORE_FOR_PICKLING
+        # to avoid runtime error.
+        for remote_module in self._create_remote_module_iter(
+            dst_worker_name, modes=[ModuleCreationMode.MODULE_CTOR]
+        ):
+            new_attr_name = "new_attr"
+            setattr(remote_module, new_attr_name, 1)
+            with self.assertRaisesRegex(
+                RuntimeError,
+                "Attribute ``{}`` of RemoteModule must be either in "
+                "``_REMOTE_MODULE_PICKLED_ATTRIBUTES``  or ``_REMOTE_MODULE_ATTRIBUTES_IGNORE_FOR_PICKLING``".format(
+                    new_attr_name
+                ),
+            ):
+                rpc.rpc_sync(
+                    dst_worker_name, remote_module_attributes, (remote_module,)
+                )
+
+
+class ThreeWorkersRemoteModuleTest(CommonRemoteModuleTest):
+    @property
+    def world_size(self):  # Override setting in CommonRemoteModuleTest
+        return 3
+
+    @dist_utils.dist_init
+    def test_send_remote_module_over_the_wire(self):
+        if self.rank != 0:
+            return
+        dst_worker1_name = dist_utils.worker_name((self.rank + 1) % self.world_size)
+        dst_worker2_name = dist_utils.worker_name((self.rank + 2) % self.world_size)
+
+        # Unpickled attribtes include both the inherent attributes of RemoteModule
+        # (not inherited from the superclass) and two installed methods.
+        expected_unpickled_attrs = list(_REMOTE_MODULE_PICKLED_ATTRIBUTES)
```

Comment:
Kind of confused on the name here, its called `expected_unpickled_attrs` but we set it to pickled attributes and the generated methods?

Is it accurate to say that this represents the methods we expect to exist on remote end?

Reply:
Kind of confused on the name here, its called expected_unpickled_attrs but we set it to pickled attributes and the generated methods?

I think "methods" can also be viewed as "attributes" in Python, since a method can be set via `setattr`. It's just a bit lengthy to call "expected_unpickled_attrs_and_methods" instead.

> Is it accurate to say that this represents the methods we expect to exist on remote end?

Exactly. So this remote node can access the attributes and the generated methods of the remote node passed from the first node.